File: src/contests/cg/puzzle/haskell/the-descent.hs
-- 🎮 CodinGame Puzzle - the-descent
-- https://www.codingame.com/training/easy/the-descent

import System.IO

-- Game loop: each turn, read 8 mountain heights and fire on the tallest one.
main :: IO ()
main = do
  hSetBuffering stdout NoBuffering
  let loop = do
        hs <- mapM (const (readLn :: IO Int)) [1 .. 8 :: Int]
        let go _ _ mi [] = mi
            go i mh mi (h : t) =
              if h > mh then go (i + 1) h i t else go (i + 1) mh mi t
        print (go (0 :: Int) (-1 :: Int) (0 :: Int) hs)
        loop
  loop

File: src/contests/cg/puzzle/pascal/the-descent.pas
// 🎮 CodinGame Puzzle - the-descent
// https://www.codingame.com/training/easy/the-descent

{$mode objfpc}{$H+}
var
  maxHeight, maxIndex, h, i: integer;
begin
  // Game loop: each turn, read 8 mountain heights and fire on the tallest one.
  while true do
  begin
    maxHeight := -1;
    maxIndex := 0;
    for i := 0 to 7 do
    begin
      readln(h);
      if h > maxHeight then
      begin
        maxHeight := h;
        maxIndex := i;
      end;
    end;
    writeln(maxIndex);
    flush(output);
  end;
end.
